packages/core/src/common/uri.ts

```diff
@@ -213,4 +214,11 @@ export default class URI {
         return result;
     }
 
+    static file(path: string): URI {
+        if (isWindows) {
+            return new URI(Uri.file(path.replace(/\\/g, '/')));
+        } else {
+            return new URI(Uri.file(path));
+        }
+    }
```

Comment:
This is the part I was unsure about, @akosyakov do you have any comments about it?

Reply:
it should be removed. There is `FileUri` and it can be used only on the backend. Please see URIs/paths guidelines: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#uripath

Reply:
Will take care